### PR TITLE
Clean up skybox

### PR DESCRIPTION
**Pull request by pmconne**
Breaking API change, but this API was already pretty "broken" anyway. Users in China - among others - want to be able to define sky boxes using images that are *not* stored in the iModel.

Previously, DisplayStyle3dSettings.environment was a mutable EnvironmentProps and was `@internal`.
DisplayStyle3dState.environment was a mutable Environment (from core-frontend).
SkyBoxProps was implemented in core-common but core-frontend produced from it one of SkyCube, SkyGradient, or SkySphere, with no relationship between them. All of those were `@internal`.
Sky cube and sky sphere required that the images used by the skybox refer to persistent texture elements.
Frequently, changing the skybox would appear to have no effect - the async loading of the skybox was buggy.

Now:
A skybox image can be a texture Id or an image URL.
Environment is immutable and contains an immutable SkyBox and GroundPlane along with flags controlling whether each is displayed.
ViewState3d has (while attached to a viewport) and EnvironmentDecorations responsible for keeping graphics for ground plane and sky box up to date.

### CODE REVIEW

**Review by bbastings — APPROVED** [review]

**Comment by pmconne** [comment]
Sky cube display appears to be broken. It has not been tested in over a year. That will have to be investigated and addressed separately.

**Review by markschlosseratbentley — APPROVED** [review]

**Review by MichaelBelousov — APPROVED** [review]

File: common/api/core-frontend.api.md [hunk 1]
```diff
@@ -9118,19 +9142,9 @@ export class SheetViewState extends ViewState2d {
...
-    // @internal (undocumented)
-    abstract loadParams(_system: RenderSystem, _iModel: IModelConnection): SkyBoxParams;
-    // (undocumented)
-    toJSON(): SkyBoxProps;
-}
-
-// @public
+// @internal
 export namespace SkyBox {
+    // @public
```
Comment (calebmshafer) [inline common/api/core-frontend.api.md:9147]: Why are these public members when the namespace itself is internal? I don't think they'd show in the docs this way.

Comment (pmconne) [reply common/api/core-frontend.api.md:9147]: They won't show in the docs, but they should not be marked public.
I left a ###TODO to clean this bit up, will address now.

Comment (pmconne) [reply common/api/core-frontend.api.md]: Addressed.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**